desktop/mac/permission-requirements.md

```diff
@@ -7,15 +7,85 @@ redirect_from:
+
+### Binding privileged ports
+
+When running a container that requires privileged ports binding, Docker Desktop first attempts to bind it directly as an unprivileged process. In case of failure due to the OS preventing it, Docker Desktop checks if the `com.docker.vmnetd` privileged helper process is running to bind the privileged port through it. If the privileged helper process is not running, Docker Desktop prompts the user for authorization to run it under [launchd](https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPSystemStartup/Chapters/CreatingLaunchdJobs.html).
+This configures the privileged helper to run as in the versions of Docker Desktop prior to v4.15. However, the functionality provided by this privileged helper has been stripped down. It now only supports port binding and caching the Registry Access Management policy.
+If the user declines the launch of the privileged helper process, the privileged port binding cannot be done and the docker CLI returns an error:
+```console
+$ docker run -p 127.0.0.1:80:80 nginx
+
+docker: Error response from daemon: Ports are not available: exposing port
+TCP 127.0.0.1:80 -> 0.0.0.0:0: failed to connect to /var/run/com.docker.vmnetd.sock:
+is vmnetd running?: dial unix /var/run/com.docker.vmnetd.sock: connect: connection
+refused.
+ERRO[0003] error waiting for container: context canceled
+```
+
+> **Note**
+>
+> The command may fail with the same error if user takes too long to authorize the prompt to start the helper process as it may timeout.
+
+The removal of the privileged helper process can be done in the typical way of removing launchd processes.
+
+```console
+$ ps aux | grep vmnetd                                                
+root             28739   0.0  0.0 34859128    228   ??  Ss    6:03PM   0:00.06 /Library/PrivilegedHelperTools/com.docker.vmnetd
+user             32222   0.0  0.0 34122828    808 s000  R+   12:55PM   0:00.00 grep vmnetd
+$ sudo launchctl unload -w /Library/LaunchDaemons/com.docker.vmnetd.plist
+Password:
+$ ps aux | grep vmnetd                                                   
+user             32242   0.0  0.0 34122828    716 s000  R+   12:55PM   0:00.00 grep vmnetd
+$ rm /Library/LaunchDaemons/com.docker.vmnetd.plist 
+$ rm /Library/PrivilegedHelperTools/com.docker.vmnetd 
+```
+
+### Ensuring `localhost` and `kubernetes.docker.internal` are defined
+
+On first run, Docker Desktop checks if `localhost` is resolved to `127.0.0.1`. In case the resolution fails, it prompts the user to allow adding the mapping to `/etc/hosts`. Similarly, when the Kubernetes cluster is installed, it checks that `kubernetes.docker.internal` is resolved to `127.0.0.1` and prompts the user to do so. 
+
+## Privileged Helper
```

Comment:
I suggest to remove the "Privileged helper" section fully, as it is confusing now that we said that _the functionality provided by this privileged helper has been stripped down. It now only supports port binding and caching the Registry Access Management policy._


Or, we can keep it, but change it to something like this (although it feels like a repetition of what we already said above):

```
In the limited situations when the privileged helper is needed (i.e. binding privileged ports or caching the Registry Access Management policy), the privileged helper is started by `launchd` and runs in the background unless it is disabled at runtime as previously described. The Docker Desktop backend communicates with it over the UNIX domain socket `/var/run/com.docker.vmnetd.sock`. The functionalities it performs are: 
- Binding privileged ports that are less than 1024.
- Securely caching the Registry Access Management policy which is read-only for the developer.
- Uninstalling the privileged helper.
```